Code/src/main/java/nl/utwente/group10/haskell/catalog/ClassEntry.java

```diff
@@ -0,0 +1,70 @@
+package nl.utwente.group10.haskell.catalog;
+
+import nl.utwente.group10.haskell.type.Type;
+import nl.utwente.group10.haskell.type.TypeClass;
+import nl.utwente.group10.haskell.typeparser.TypeBuilder;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/** A class entry in the Haskell catalog. */
+public class ClassEntry implements Comparable<ClassEntry> {
```

Comment:
Could perhaps use a common superclass `CatalogEntry` - they're both named *somethings* that get sorted on their names.